// @status/codes/lib/braintree.js
"use strict";
// https://articles.braintreepayments.com/control-panel/transactions/declines#authorization-decline-codes
Object.defineProperty(exports, "__esModule", { value: true });
var Braintree;
(function (Braintree) {
    Braintree[Braintree["DoNotHonor"] = 2000] = "DoNotHonor";
    Braintree[Braintree["InsufficientFunds"] = 2001] = "InsufficientFunds";
    Braintree[Braintree["LimitExceeded"] = 2002] = "LimitExceeded";
    Braintree[Braintree["CardholdersActivityLimitExceeded"] = 2003] = "CardholdersActivityLimitExceeded";
    Braintree[Braintree["ExpiredCard"] = 2004] = "ExpiredCard";
    Braintree[Braintree["InvalidCreditCardNumber"] = 2005] = "InvalidCreditCardNumber";
    Braintree[Braintree["InvalidExpirationDate"] = 2006] = "InvalidExpirationDate";
    Braintree[Braintree["NoAccount"] = 2007] = "NoAccount";
    Braintree[Braintree["CreditCardLengthError"] = 2008] = "CreditCardLengthError";
    Braintree[Braintree["NoSuchIssuer"] = 2009] = "NoSuchIssuer";
    Braintree[Braintree["CardIssuerDeclinedCVV"] = 2010] = "CardIssuerDeclinedCVV";
    Braintree[Braintree["VoiceAuthorizationRequired"] = 2011] = "VoiceAuthorizationRequired";
    Braintree[Braintree["ProcessorDeclinedPossibleLostCard"] = 2012] = "ProcessorDeclinedPossibleLostCard";
    Braintree[Braintree["ProcessorDeclinedPossibleStolenCard"] = 2013] = "ProcessorDeclinedPossibleStolenCard";
    Braintree[Braintree["ProcessorDeclinedFraudSuspected"] = 2014] = "ProcessorDeclinedFraudSuspected";
    Braintree[Braintree["TransactionNotAllowed"] = 2015] = "TransactionNotAllowed";
    Braintree[Braintree["DuplicateTransaction"] = 2016] = "DuplicateTransaction";
    Braintree[Braintree["CardholderStoppedBilling"] = 2017] = "CardholderStoppedBilling";
    Braintree[Braintree["CardholderStoppedAllBilling"] = 2018] = "CardholderStoppedAllBilling";
    Braintree[Braintree["InvalidTransaction"] = 2019] = "InvalidTransaction";
    Braintree[Braintree["Violation"] = 2020] = "Violation";
    Braintree[Braintree["SecurityViolation"] = 2021] = "SecurityViolation";
    Braintree[Braintree["DeclinedUpdatedCardholderAvailable"] = 2022] = "DeclinedUpdatedCardholderAvailable";
    Braintree[Braintree["ProcessorDoesNotSupportThisFeature"] = 2023] = "ProcessorDoesNotSupportThisFeature";
    Braintree[Braintree["CardTypeNotEnabled"] = 2024] = "CardTypeNotEnabled";
    Braintree[Braintree["SetUpErrorMerchant"] = 2025] = "SetUpErrorMerchant";
    Braintree[Braintree["InvalidMerchantID"] = 2026] = "InvalidMerchantID";
    Braintree[Braintree["SetUpErrorAmount"] = 2027] = "SetUpErrorAmount";
    Braintree[Braintree["SetUpErrorHierarchy"] = 2028] = "SetUpErrorHierarchy";
    Braintree[Braintree["SetUpErrorCard"] = 2029] = "SetUpErrorCard";
    Braintree[Braintree["SetUpErrorTerminal"] = 2030] = "SetUpErrorTerminal";
    Braintree[Braintree["EncryptionError"] = 2031] = "EncryptionError";
    Braintree[Braintree["SurchargeNotPermitted"] = 2032] = "SurchargeNotPermitted";
    Braintree[Braintree["InconsistentData"] = 2033] = "InconsistentData";
    Braintree[Braintree["NoActionTaken"] = 2034] = "NoActionTaken";
    Braintree[Braintree["PartialApprovalForAmountInGroupIIIVersion"] = 2035] = "PartialApprovalForAmountInGroupIIIVersion";
    Braintree[Braintree["AuthorizationCouldNotBeFound"] = 2036] = "AuthorizationCouldNotBeFound";
    Braintree[Braintree["AlreadyReversed"] = 2037] = "AlreadyReversed";
    Braintree[Braintree["ProcessorDeclined"] = 2038] = "ProcessorDeclined";
    Braintree[Braintree["InvalidAuthorizationCode"] = 2039] = "InvalidAuthorizationCode";
    Braintree[Braintree["InvalidStore"] = 2040] = "InvalidStore";
    Braintree[Braintree["DeclinedCallForApproval"] = 2041] = "DeclinedCallForApproval";
    Braintree[Braintree["InvalidClientID"] = 2042] = "InvalidClientID";
    Braintree[Braintree["ErrorDoNotRetryCallIssuer"] = 2043] = "ErrorDoNotRetryCallIssuer";
    Braintree[Braintree["DeclinedCallIssuer"] = 2044] = "DeclinedCallIssuer";
    Braintree[Braintree["InvalidMerchantNumber"] = 2045] = "InvalidMerchantNumber";
    Braintree[Braintree["Declined"] = 2046] = "Declined";
    Braintree[Braintree["CallIssuerPickUpCard"] = 2047] = "CallIssuerPickUpCard";
    Braintree[Braintree["InvalidAmount"] = 2048] = "InvalidAmount";
    Braintree[Braintree["InvalidSkuNumber"] = 2049] = "InvalidSkuNumber";
    Braintree[Braintree["InvalidCreditPlan"] = 2050] = "InvalidCreditPlan";
    Braintree[Braintree["CreditCardNumberDoesNotMatchMethodOfPayment"] = 2051] = "CreditCardNumberDoesNotMatchMethodOfPayment";
    Braintree[Braintree["CardReportedAsLostOrStolen"] = 2053] = "CardReportedAsLostOrStolen";
    Braintree[Braintree["ReversalAmountDoesNotMatchAuthorizationAmount"] = 2054] = "ReversalAmountDoesNotMatchAuthorizationAmount";
    Braintree[Braintree["InvalidTransactionDivisionNumber"] = 2055] = "InvalidTransactionDivisionNumber";
    Braintree[Braintree["TransactionAmountExceedsTheTransactionDivisionLimit"] = 2056] = "TransactionAmountExceedsTheTransactionDivisionLimit";
    Braintree[Braintree["IssuerOrCardholderHasPutARestrictionOnTheCard"] = 2057] = "IssuerOrCardholderHasPutARestrictionOnTheCard";
    Braintree[Braintree["MerchantNotMastercardSecurecodeEnabled"] = 2058] = "MerchantNotMastercardSecurecodeEnabled";
    Braintree[Braintree["AddressVerificationFailed"] = 2059] = "AddressVerificationFailed";
    Braintree[Braintree["AddressVerificationAndCardSecurityCodeFailed"] = 2060] = "AddressVerificationAndCardSecurityCodeFailed";
    Braintree[Braintree["InvalidTransactionData"] = 2061] = "InvalidTransactionData";
    Braintree[Braintree["InvalidTaxAmount"] = 2062] = "InvalidTaxAmount";
    Braintree[Braintree["PaypalBusinessAccountPreferenceResultedInTheTransactionFailing"] = 2063] = "PaypalBusinessAccountPreferenceResultedInTheTransactionFailing";
    Braintree[Braintree["InvalidCurrencyCode"] = 2064] = "InvalidCurrencyCode";
    Braintree[Braintree["RefundTimeLimitExceeded"] = 2065] = "RefundTimeLimitExceeded";
    Braintree[Braintree["PaypalBusinessAccountRestricted"] = 2066] = "PaypalBusinessAccountRestricted";
    Braintree[Braintree["AuthorizationExpired"] = 2067] = "AuthorizationExpired";
    Braintree[Braintree["PaypalBusinessAccountLockedOrClosed"] = 2068] = "PaypalBusinessAccountLockedOrClosed";
    Braintree[Braintree["PaypalBlockingDuplicateOrderIds"] = 2069] = "PaypalBlockingDuplicateOrderIds";
    Braintree[Braintree["PaypalBuyerRevokedFuturePaymentAuthorization"] = 2070] = "PaypalBuyerRevokedFuturePaymentAuthorization";
    Braintree[Braintree["PaypalPayeeAccountInvalidOrDoesNotHaveAConfirmedEmail"] = 2071] = "PaypalPayeeAccountInvalidOrDoesNotHaveAConfirmedEmail";
    Braintree[Braintree["PaypalPayeeEmailIncorrectlyFormatted"] = 2072] = "PaypalPayeeEmailIncorrectlyFormatted";
    Braintree[Braintree["PaypalValidationError"] = 2073] = "PaypalValidationError";
    Braintree[Braintree["FundingInstrumentInThePaypalAccountWasDeclinedByTheProcessorOrBank"] = 2074] = "FundingInstrumentInThePaypalAccountWasDeclinedByTheProcessorOrBank";
    Braintree[Braintree["PayerAccountIsLockedOrClosed"] = 2075] = "PayerAccountIsLockedOrClosed";
    Braintree[Braintree["PayerCannotPayForThisTransactionWithPaypal"] = 2076] = "PayerCannotPayForThisTransactionWithPaypal";
    Braintree[Braintree["TransactionRefusedDueToPaypalRiskModel"] = 2077] = "TransactionRefusedDueToPaypalRiskModel";
    Braintree[Braintree["PaypalMerchantAccountConfigurationError"] = 2079] = "PaypalMerchantAccountConfigurationError";
    Braintree[Braintree["PaypalPendingPaymentsAreNotSupported"] = 2081] = "PaypalPendingPaymentsAreNotSupported";
    Braintree[Braintree["PaypalDomesticTransactionRequired"] = 2082] = "PaypalDomesticTransactionRequired";
    Braintree[Braintree["PaypalPhoneNumberRequired"] = 2083] = "PaypalPhoneNumberRequired";
    Braintree[Braintree["PaypalTaxInfoRequired"] = 2084] = "PaypalTaxInfoRequired";
    Braintree[Braintree["PaypalPayeeBlockedTransaction"] = 2085] = "PaypalPayeeBlockedTransaction";
    Braintree[Braintree["PaypalTransactionLimitExceeded"] = 2086] = "PaypalTransactionLimitExceeded";
    Braintree[Braintree["PaypalReferenceTransactionsNotEnabledForYourAccount"] = 2087] = "PaypalReferenceTransactionsNotEnabledForYourAccount";
    Braintree[Braintree["CurrencyNotEnabledForYourPaypalSellerAccount"] = 2088] = "CurrencyNotEnabledForYourPaypalSellerAccount";
    Braintree[Braintree["PaypalPayeeEmailPermissionDeniedForThisRequest"] = 2089] = "PaypalPayeeEmailPermissionDeniedForThisRequest";
    Braintree[Braintree["PaypalAccountNotConfiguredToRefundMoreThanSettledAmount"] = 2090] = "PaypalAccountNotConfiguredToRefundMoreThanSettledAmount";
    Braintree[Braintree["CurrencyOfThisTransactionMustMatchCurrencyOfYourPaypalAccount"] = 2091] = "CurrencyOfThisTransactionMustMatchCurrencyOfYourPaypalAccount";
    Braintree[Braintree["NoDataFoundTryAnotherVerificationMethod"] = 2092] = "NoDataFoundTryAnotherVerificationMethod";
    Braintree[Braintree["ProcessorDeclinedRangeStart"] = 2093] = "ProcessorDeclinedRangeStart";
    Braintree[Braintree["ProcessorDeclinedRangeEnd"] = 2999] = "ProcessorDeclinedRangeEnd";
    Braintree[Braintree["ProcessorNetworkUnavailableTryAgain"] = 3000] = "ProcessorNetworkUnavailableTryAgain";
    Braintree[Braintree["SettlementDeclined"] = 4001] = "SettlementDeclined";
    Braintree[Braintree["AlreadyCaptured"] = 4003] = "AlreadyCaptured";
    Braintree[Braintree["AlreadyRefunded"] = 4004] = "AlreadyRefunded";
    Braintree[Braintree["CaptureAmountExceededAllowableLimit"] = 4006] = "CaptureAmountExceededAllowableLimit";
    Braintree[Braintree["PaypalPendingPaymentsNotSupported"] = 4018] = "PaypalPendingPaymentsNotSupported";
})(Braintree = exports.Braintree || (exports.Braintree = {}));
