// @status/codes/lib/cloudflare.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var helpers_1 = require("./helpers");
var http_1 = require("./http");
var Cloudflare;
(function (Cloudflare) {
    Cloudflare[Cloudflare["UnknownError"] = 520] = "UnknownError";
    Cloudflare[Cloudflare["WebServerIsDown"] = 521] = "WebServerIsDown";
    Cloudflare[Cloudflare["ConnectionTimedOut"] = 522] = "ConnectionTimedOut";
    Cloudflare[Cloudflare["OriginIsUnreachable"] = 523] = "OriginIsUnreachable";
    Cloudflare[Cloudflare["ATimeoutOccurred"] = 524] = "ATimeoutOccurred";
    Cloudflare[Cloudflare["SslHandshakeFailed"] = 525] = "SslHandshakeFailed";
    Cloudflare[Cloudflare["InvalidSslCertificate"] = 526] = "InvalidSslCertificate";
    Cloudflare[Cloudflare["RailgunError"] = 527] = "RailgunError";
    Cloudflare[Cloudflare["OriginDnsError"] = 530] = "OriginDnsError";
})(Cloudflare = exports.Cloudflare || (exports.Cloudflare = {}));
exports.CloudflareHttp = helpers_1.mergeEnums(http_1.Http, Cloudflare);
