// @status/codes/lib/firebase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var FirebaseAuth;
(function (FirebaseAuth) {
    FirebaseAuth["AppDeleted"] = "auth/app-deleted";
    FirebaseAuth["NotAuthorized"] = "auth/app-not-authorized";
    FirebaseAuth["ArgumentError"] = "auth/argument-error";
    FirebaseAuth["InvalidApiKey"] = "auth/invalid-api-key";
    FirebaseAuth["InvalidUserToken"] = "auth/invalid-user-token";
    FirebaseAuth["NetworkRequestFailed"] = "auth/network-request-failed";
    FirebaseAuth["OperationNotAllowed"] = "auth/operation-not-allowed";
    FirebaseAuth["RequiresRecentLogin"] = "auth/requires-recent-login";
    FirebaseAuth["TooManyRequests"] = "auth/too-many-requests";
    FirebaseAuth["UnauthorizedDomain"] = "auth/unauthorized-domain";
    FirebaseAuth["UserDisabled"] = "auth/user-disabled";
    FirebaseAuth["UserTokenExpired"] = "auth/user-token-expired";
    FirebaseAuth["WebStorageUnsupported"] = "auth/web-storage-unsupported";
})(FirebaseAuth = exports.FirebaseAuth || (exports.FirebaseAuth = {}));
