// @status/codes/lib/ftp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var FTP;
(function (FTP) {
    FTP[FTP["RestartMarkerReplay"] = 110] = "RestartMarkerReplay";
    FTP[FTP["ServiceReadyInNnnMinutes"] = 120] = "ServiceReadyInNnnMinutes";
    FTP[FTP["DataConnectionAlreadyOpenTransferStarting"] = 125] = "DataConnectionAlreadyOpenTransferStarting";
    FTP[FTP["FileStatusOkayAboutToOpenDataConnection"] = 150] = "FileStatusOkayAboutToOpenDataConnection";
    FTP[FTP["CommandNotImplementedSuperfluous"] = 202] = "CommandNotImplementedSuperfluous";
    FTP[FTP["SystemStatus"] = 211] = "SystemStatus";
    FTP[FTP["DirectoryStatus"] = 212] = "DirectoryStatus";
    FTP[FTP["FileStatus"] = 213] = "FileStatus";
    FTP[FTP["HelpMessage"] = 214] = "HelpMessage";
    FTP[FTP["NameSystemType"] = 215] = "NameSystemType";
    FTP[FTP["ServiceReadyForNewUser"] = 220] = "ServiceReadyForNewUser";
    FTP[FTP["ServiceClosingControlConnection"] = 221] = "ServiceClosingControlConnection";
    FTP[FTP["DataConnectionOpenNoTransferInProgress"] = 225] = "DataConnectionOpenNoTransferInProgress";
    FTP[FTP["ClosingDataConnectionRequestedFileActionSuccess"] = 226] = "ClosingDataConnectionRequestedFileActionSuccess";
    FTP[FTP["EnteringPassiveMode"] = 227] = "EnteringPassiveMode";
    FTP[FTP["EnteringLongPassiveMode"] = 228] = "EnteringLongPassiveMode";
    FTP[FTP["EnteringExtendedPassiveMode"] = 229] = "EnteringExtendedPassiveMode";
    FTP[FTP["UserLoggedInProceed"] = 230] = "UserLoggedInProceed";
    FTP[FTP["UserLoggedOutServiceTerminated"] = 231] = "UserLoggedOutServiceTerminated";
    FTP[FTP["LogoutCommandNotedWillCompleteWhenTransferDone"] = 232] = "LogoutCommandNotedWillCompleteWhenTransferDone";
    // A higher level nonstandard code created by Microsoft.
    FTP[FTP["AuthenticationMechanismAccepted"] = 234] = "AuthenticationMechanismAccepted";
    FTP[FTP["RequestedFileActionOkayCompleted"] = 250] = "RequestedFileActionOkayCompleted";
    FTP[FTP["PathnameCreated"] = 257] = "PathnameCreated";
    FTP[FTP["UserNameOkayNeedPassword"] = 331] = "UserNameOkayNeedPassword";
    FTP[FTP["NeedAccountForLogin"] = 332] = "NeedAccountForLogin";
    FTP[FTP["RequestedFileActionPendingFurtherInformation"] = 350] = "RequestedFileActionPendingFurtherInformation";
    FTP[FTP["ServiceNotAvailableClosingControlConnection"] = 421] = "ServiceNotAvailableClosingControlConnection";
    FTP[FTP["CannotOpenDataConnection"] = 425] = "CannotOpenDataConnection";
    FTP[FTP["ConnectionClosedTransferAborted"] = 426] = "ConnectionClosedTransferAborted";
    FTP[FTP["InvalidUsernameOrPassword"] = 430] = "InvalidUsernameOrPassword";
    FTP[FTP["RequestedHostUnavailable"] = 434] = "RequestedHostUnavailable";
    FTP[FTP["RequestedFileActionNotTaken"] = 450] = "RequestedFileActionNotTaken";
    FTP[FTP["RequestedActionAbortedLocalErrorInProcessing"] = 451] = "RequestedActionAbortedLocalErrorInProcessing";
    FTP[FTP["RequestedActionNotTakenInsufficientStorage"] = 452] = "RequestedActionNotTakenInsufficientStorage";
    FTP[FTP["SyntaxErrorCommandUnrecognizedCommandLineTooLong"] = 500] = "SyntaxErrorCommandUnrecognizedCommandLineTooLong";
    FTP[FTP["SyntaxErrorInParametersOrArguments"] = 501] = "SyntaxErrorInParametersOrArguments";
    FTP[FTP["CommandNotImplemented"] = 502] = "CommandNotImplemented";
    FTP[FTP["BadSequenceOfCommands"] = 503] = "BadSequenceOfCommands";
    FTP[FTP["CommandNotImplementedForThatParameter"] = 504] = "CommandNotImplementedForThatParameter";
    FTP[FTP["NotLoggedIn"] = 530] = "NotLoggedIn";
    FTP[FTP["NeedAccountForStoringFiles"] = 532] = "NeedAccountForStoringFiles";
    FTP[FTP["CouldNotConnectToServerPolicyRequiresSsl"] = 534] = "CouldNotConnectToServerPolicyRequiresSsl";
    FTP[FTP["RequestedActionNotTakenFileUnavailable"] = 550] = "RequestedActionNotTakenFileUnavailable";
    FTP[FTP["RequestedActionAbortedPageTypeUnknown"] = 551] = "RequestedActionAbortedPageTypeUnknown";
    FTP[FTP["RequestedFileActionAbortedExceededStorageAllocation"] = 552] = "RequestedFileActionAbortedExceededStorageAllocation";
    FTP[FTP["RequestedActionNotTakenFileNameNotAllowed"] = 553] = "RequestedActionNotTakenFileNameNotAllowed";
    FTP[FTP["IntegrityProtectedReply"] = 631] = "IntegrityProtectedReply";
    FTP[FTP["ConfidentialityAndIntegrityProtectedReply"] = 632] = "ConfidentialityAndIntegrityProtectedReply";
    FTP[FTP["ConfidentialityProtectedReply"] = 633] = "ConfidentialityProtectedReply";
    FTP[FTP["ConnectionResetByPeerForciblyClosedByRemoteHost"] = 10054] = "ConnectionResetByPeerForciblyClosedByRemoteHost";
    FTP[FTP["CannotConnectToRemoteServer"] = 10060] = "CannotConnectToRemoteServer";
    FTP[FTP["CannotConnectToRemoteServerActivelyRefusedByServer"] = 10061] = "CannotConnectToRemoteServerActivelyRefusedByServer";
    FTP[FTP["DirectoryNotEmpty"] = 10066] = "DirectoryNotEmpty";
    FTP[FTP["TooManyUsersServerIsFull"] = 10068] = "TooManyUsersServerIsFull";
})(FTP = exports.FTP || (exports.FTP = {}));
