// @status/codes/lib/twitter.js
"use strict";
// https://developer.twitter.com/en/docs/basics/response-codes
Object.defineProperty(exports, "__esModule", { value: true });
var TwitterHttp;
(function (TwitterHttp) {
    TwitterHttp[TwitterHttp["Ok"] = 200] = "Ok";
    TwitterHttp[TwitterHttp["NotModified"] = 304] = "NotModified";
    TwitterHttp[TwitterHttp["BadRequest"] = 400] = "BadRequest";
    TwitterHttp[TwitterHttp["Unauthorized"] = 401] = "Unauthorized";
    TwitterHttp[TwitterHttp["Forbidden"] = 403] = "Forbidden";
    TwitterHttp[TwitterHttp["NotFound"] = 404] = "NotFound";
    TwitterHttp[TwitterHttp["NotAcceptable"] = 406] = "NotAcceptable";
    TwitterHttp[TwitterHttp["Gone"] = 410] = "Gone";
    TwitterHttp[TwitterHttp["EnhanceYourCalm"] = 420] = "EnhanceYourCalm";
    TwitterHttp[TwitterHttp["UnprocessableEntity"] = 422] = "UnprocessableEntity";
    TwitterHttp[TwitterHttp["TooManyRequests"] = 429] = "TooManyRequests";
    TwitterHttp[TwitterHttp["InternalServerError"] = 500] = "InternalServerError";
    TwitterHttp[TwitterHttp["BadGateway"] = 502] = "BadGateway";
    TwitterHttp[TwitterHttp["ServiceUnavailable"] = 503] = "ServiceUnavailable";
    TwitterHttp[TwitterHttp["GatewayTimeout"] = 504] = "GatewayTimeout";
})(TwitterHttp = exports.TwitterHttp || (exports.TwitterHttp = {}));
var TwitterErrors;
(function (TwitterErrors) {
    TwitterErrors[TwitterErrors["InvalidCoordinates"] = 3] = "InvalidCoordinates";
    TwitterErrors[TwitterErrors["NoIpLocationAssociation"] = 13] = "NoIpLocationAssociation";
    TwitterErrors[TwitterErrors["NoUserTermsMatch"] = 17] = "NoUserTermsMatch";
    TwitterErrors[TwitterErrors["AuthenticationFailure"] = 32] = "AuthenticationFailure";
    TwitterErrors[TwitterErrors["PageDoesNotExist"] = 34] = "PageDoesNotExist";
    TwitterErrors[TwitterErrors["AttachmentUrlParameterInvalid"] = 44] = "AttachmentUrlParameterInvalid";
    TwitterErrors[TwitterErrors["UserNotFound"] = 50] = "UserNotFound";
    TwitterErrors[TwitterErrors["UserSuspended"] = 63] = "UserSuspended";
    TwitterErrors[TwitterErrors["AccountSuspended"] = 64] = "AccountSuspended";
    TwitterErrors[TwitterErrors["TwitterAPIv1Inactive"] = 68] = "TwitterAPIv1Inactive";
    TwitterErrors[TwitterErrors["ClientActionNotPermitted"] = 87] = "ClientActionNotPermitted";
    TwitterErrors[TwitterErrors["RateLimitExceeded"] = 88] = "RateLimitExceeded";
    TwitterErrors[TwitterErrors["InvalidOrExpiredToken"] = 89] = "InvalidOrExpiredToken";
    TwitterErrors[TwitterErrors["SslRequired"] = 92] = "SslRequired";
    TwitterErrors[TwitterErrors["DirectMessageAccessDenied"] = 93] = "DirectMessageAccessDenied";
    TwitterErrors[TwitterErrors["CredentialVerificationFailure"] = 99] = "CredentialVerificationFailure";
    TwitterErrors[TwitterErrors["AccountUpdateFailed"] = 120] = "AccountUpdateFailed";
    TwitterErrors[TwitterErrors["OverCapacity"] = 130] = "OverCapacity";
    TwitterErrors[TwitterErrors["InternalError"] = 131] = "InternalError";
    TwitterErrors[TwitterErrors["CouldNotAuthenticate"] = 135] = "CouldNotAuthenticate";
    TwitterErrors[TwitterErrors["AlreadyFavorited"] = 139] = "AlreadyFavorited";
    TwitterErrors[TwitterErrors["NoStatusFound"] = 144] = "NoStatusFound";
    TwitterErrors[TwitterErrors["UserMessageSendFailure"] = 150] = "UserMessageSendFailure";
    TwitterErrors[TwitterErrors["MessageSendError"] = 151] = "MessageSendError";
    TwitterErrors[TwitterErrors["DuplicateFollowRequest"] = 160] = "DuplicateFollowRequest";
    TwitterErrors[TwitterErrors["FollowLimitExceeded"] = 161] = "FollowLimitExceeded";
    TwitterErrors[TwitterErrors["UnauthorizedStatusRequest"] = 179] = "UnauthorizedStatusRequest";
    TwitterErrors[TwitterErrors["DailyUpdateLimitExceeded"] = 185] = "DailyUpdateLimitExceeded";
    TwitterErrors[TwitterErrors["TweetTextOverLimit"] = 186] = "TweetTextOverLimit";
    TwitterErrors[TwitterErrors["DuplicateStatus"] = 187] = "DuplicateStatus";
    TwitterErrors[TwitterErrors["SpamReportLimitExceeded"] = 205] = "SpamReportLimitExceeded";
    TwitterErrors[TwitterErrors["BadAuthenticationData"] = 215] = "BadAuthenticationData";
    TwitterErrors[TwitterErrors["UnauthorizedResourceCredentials"] = 220] = "UnauthorizedResourceCredentials";
    TwitterErrors[TwitterErrors["PossibleAutomatedRequest"] = 226] = "PossibleAutomatedRequest";
    TwitterErrors[TwitterErrors["UserMustVerifyLogin"] = 231] = "UserMustVerifyLogin";
    TwitterErrors[TwitterErrors["EndpointRetired"] = 251] = "EndpointRetired";
    TwitterErrors[TwitterErrors["CannotPerformWriteActions"] = 261] = "CannotPerformWriteActions";
    TwitterErrors[TwitterErrors["SelfMuteNotAllowed"] = 271] = "SelfMuteNotAllowed";
    TwitterErrors[TwitterErrors["NotMutingSpecifiedUser"] = 272] = "NotMutingSpecifiedUser";
    TwitterErrors[TwitterErrors["SingleAnimatedGifAllowed"] = 323] = "SingleAnimatedGifAllowed";
    TwitterErrors[TwitterErrors["MediaIdsValidationFailure"] = 324] = "MediaIdsValidationFailure";
    TwitterErrors[TwitterErrors["MediaIdNotFound"] = 325] = "MediaIdNotFound";
    TwitterErrors[TwitterErrors["AccountTemporarilyLocked"] = 326] = "AccountTemporarilyLocked";
    TwitterErrors[TwitterErrors["AlreadyRetweeted"] = 327] = "AlreadyRetweeted";
    TwitterErrors[TwitterErrors["DirectMessageCharacterLimitExceeded"] = 354] = "DirectMessageCharacterLimitExceeded";
    TwitterErrors[TwitterErrors["ReplyToUnavailableTweet"] = 385] = "ReplyToUnavailableTweet";
    TwitterErrors[TwitterErrors["AttachmentTypeLimitExceeded"] = 386] = "AttachmentTypeLimitExceeded";
    TwitterErrors[TwitterErrors["UrlInvalid"] = 407] = "UrlInvalid";
    TwitterErrors[TwitterErrors["CallbackUrlNotApproved"] = 415] = "CallbackUrlNotApproved";
    TwitterErrors[TwitterErrors["InvalidSuspendedApplication"] = 416] = "InvalidSuspendedApplication";
    TwitterErrors[TwitterErrors["OobOnlyOauthSupport"] = 417] = "OobOnlyOauthSupport";
})(TwitterErrors = exports.TwitterErrors || (exports.TwitterErrors = {}));
